<a href="https://colab.research.google.com/github/CristianoDataScience/Analise-Estat-stica-Enem_2019/blob/main/C%C3%B3pia_de_Tratamento_enem_2019.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##### Este projeto analise os resultados do ENEM no estado de São Paulo do ano de 2019.

##### Os dados foram extraídos do site do INEP:

<http://inep.gov.br/microdados>

### Importando Bibliotecas 

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
dados = pd.read_csv('/content/drive/MyDrive/curso estatistica/microdados_enem_2019_sp.csv',
                    sep=';', encoding='iso-8859-1')
# encoding: codificação de caracteres, normalmente utiliza-se o iso-8859-1, utf-8, latin-1

### Extração e Limpeza dos Dados

In [ ]:
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  ...  NU_NOTA_MT TP_LINGUA  \
0            3                 1  ...      6002.0         0   
1            3                 1  ...         NaN         1   
2            3                 1  ...      7318.0         0   
3            3                 1  ...      3932.0         1   
4            1                 1  ...         NaN         1   

   TP_STATUS_REDACAO NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0         160.0          200.0          180.0   
1                NaN           NaN            NaN            NaN   
2                1.0         160.0          120.0          200.0   
3                1.0          80.0          160.0          120.0   
4                NaN           NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          200.0          200.0            940.0     B  
1            NaN            NaN              NaN     A  
2          200.0          200.0            880.0     A  
3          100.0          100.0            560.0     B  
4            NaN            NaN              NaN     B  

[5 rows x 44 columns]

In [ ]:
# O número total de inscritos em São Paulo foi de 813772, sendo de 15,97% de todo o Brasil que foi de 5095308 inscritos.
dados.shape

(813772, 44)

In [ ]:
# Analise do tipo de dados 
dados.dtypes

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

In [ ]:
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  CO_UF_RESIDENCIA SG_UF_RESIDENCIA  \
0     1,90E+11            Santa Isabel                35               SP   
1     1,90E+11             Pinhalzinho                35               SP   
2     1,90E+11               São Paulo                35               SP   
3     1,90E+11   São Bernardo do Campo                35               SP   
4     1,90E+11    São Joaquim da Barra                35               SP   

   NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  TP_COR_RACA  TP_NACIONALIDADE  \
0        22       M                1            3                 1   
1        19       M                1            3                 1   
2        17       F                1            3                 1   
3        19       M                1            3                 1   
4        37       M                2            1                 1   

   CO_MUNICIPIO_NASCIMENTO  ... NU_NOTA_MT  TP_LINGUA TP_STATUS_REDACAO  \
0                1506500.0  ...     6002.0          0               1.0   
1                1503804.0  ...        NaN          1               NaN   
2                1501402.0  ...     7318.0          0               1.0   
3                3548708.0  ...     3932.0          1               1.0   
4                3549409.0  ...        NaN          1               NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   
1            NaN            NaN            NaN            NaN            NaN   
2          160.0          120.0          200.0          200.0          200.0   
3           80.0          160.0          120.0          100.0          100.0   
4            NaN            NaN            NaN            NaN            NaN   

   NU_NOTA_REDACAO  Q025  
0            940.0     B  
1              NaN     A  
2            880.0     A  
3            560.0     B  
4              NaN     B  

[5 rows x 43 columns]

In [ ]:
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA', 'SG_UF_RESIDENCIA', 'CO_MUNICIPIO_NASCIMENTO'],)

In [ ]:
dados1.head(10)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   
5     1,90E+11              Embu-Guaçu        17       F                1   
6     1,90E+11     São José dos Campos        28       F                1   
7     1,90E+11     São José dos Campos        25       F                2   
8     1,90E+11               Guarulhos        23       F                1   
9     1,90E+11               São Paulo        19       F                1   

   TP_COR_RACA  TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO  \
0            3                 1    Santa Izabel do Pará              15.0   
1            3                 1                 Jacundá              15.0   
2            3                 1                   Belém              15.0   
3            3                 1   São Bernardo do Campo              35.0   
4            1                 1    São Joaquim da Barra              35.0   
5            3                 1             Itacoatiara              13.0   
6            1                 1                  Manaus              13.0   
7            1                 1     São José dos Campos              35.0   
8            1                 1               São Paulo              35.0   
9            3                 1                   Belém              15.0   

  SG_UF_NASCIMENTO  ...  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0               PA  ...      6002.0          0                1.0   
1               PA  ...         NaN          1                NaN   
2               PA  ...      7318.0          0                1.0   
3               SP  ...      3932.0          1                1.0   
4               SP  ...         NaN          1                NaN   
5               AM  ...      4192.0          0                1.0   
6               AM  ...         NaN          0                NaN   
7               SP  ...         NaN          1                NaN   
8               SP  ...      7961.0          0                1.0   
9               PA  ...         NaN          0                NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4 NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0         200.0   
1            NaN            NaN            NaN            NaN           NaN   
2          160.0          120.0          200.0          200.0         200.0   
3           80.0          160.0          120.0          100.0         100.0   
4            NaN            NaN            NaN            NaN           NaN   
5          120.0          120.0          120.0          120.0          80.0   
6            NaN            NaN            NaN            NaN           NaN   
7            NaN            NaN            NaN            NaN           NaN   
8          160.0          200.0          180.0          180.0         160.0   
9            NaN            NaN            NaN            NaN           NaN   

   NU_NOTA_REDACAO Q025  
0            940.0    B  
1              NaN    A  
2            880.0    A  
3            560.0    B  
4              NaN    B  
5            560.0    B  
6              NaN    A  
7              NaN    B  
8            880.0    B  
9              NaN    B  

[10 rows x 40 columns]

In [ ]:
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO', 'CO_UF_NASCIMENTO',
                              'SG_UF_NASCIMENTO', 'TP_ANO_CONCLUIU', 'TP_ENSINO',
                              'CO_MUNICIPIO_ESC', 'CO_MUNICIPIO_ESC', 'CO_UF_ESC',
                              'SG_UF_ESC'],)

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 32 columns]

In [ ]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC', 'TP_LOCALIZACAO_ESC', 'TP_SIT_FUNC_ESC'],)

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

In [ ]:
dados1.shape

(813772, 29)

##### Corrigindo erros nas notas 

In [ ]:
dados1.loc[:,'NU_NOTA_CN'] /= 10

In [ ]:
dados1.head(12)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   

    TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0             3                 1                1          1             0   
1             3                 1                1          1             0   
2             3                 1                2          3             0   
3             3                 1                2          2             0   
4             1                 1                1          1             0   
5             3                 1                2          2             0   
6             1                 1                1          1             0   
7             1                 1                1          1             0   
8             1                 1                1          1             0   
9             3                 1                1          1             0   
10            1                 1                2          2             0   
11            4                 1                1          1             0   

    ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  \
0   ...      6002.0         0                1.0          160.0   
1   ...         NaN         1                NaN            NaN   
2   ...      7318.0         0                1.0          160.0   
3   ...      3932.0         1                1.0           80.0   
4   ...         NaN         1                NaN            NaN   
5   ...      4192.0         0                1.0          120.0   
6   ...         NaN         0                NaN            NaN   
7   ...         NaN         1                NaN            NaN   
8   ...      7961.0         0                1.0          160.0   
9   ...         NaN         0                NaN            NaN   
10  ...      6128.0         1                1.0          120.0   
11  ...         NaN         1                NaN            NaN   

    NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0           200.0          180.0          200.0          200.0   
1             NaN            NaN            NaN            NaN   
2           120.0          200.0          200.0          200.0   
3           160.0          120.0          100.0          100.0   
4             NaN            NaN            NaN            NaN   
5           120.0          120.0          120.0           80.0   
6             NaN            NaN            NaN            NaN   
7             NaN            NaN            NaN            NaN   
8           200.0          180.0          180.0          160.0   
9             NaN            NaN            NaN            NaN   
10          160.0          120.0           80.0          100.0   
11            NaN            NaN            NaN            NaN   

    NU_NOTA_REDACAO  Q025  
0             940.0     B  
1               NaN     A  
2             880.0     A  
3             560.0     B  
4               NaN     B  
5             560.0     B  
6               NaN     A  
7    

In [ ]:
dados1.loc[:,'NU_NOTA_CH'] /= 10
dados1.loc[:,'NU_NOTA_LC'] /= 10
dados1.loc[:,'NU_NOTA_MT'] /= 10

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

##### Removendo Variáveis (Colunas)

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NOTA_REDACAO  Q025  
0          180.0          200.0          200.0         940.0     B  
1            NaN            NaN            NaN           NaN     A  
2          200.0          200.0          200.0         880.0     A  
3          120.0          100.0          100.0         560.0     B  
4            NaN            NaN            NaN           NaN     B  

[5 rows x 29 columns]

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN', 'NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC', 'NU_NOTA_MT': 'NOTA_MT'})

In [ ]:
dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1', 'NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3', 'NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5': 'COMP5'})

In [ ]:
dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE', 'TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA', 'Q025': 'INTERNET',
                                'TP_ESCOLA': 'ESCOLA'})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL  RACA  \
0     1,90E+11            Santa Isabel     22    M                1     3   
1     1,90E+11             Pinhalzinho     19    M                1     3   
2     1,90E+11               São Paulo     17    F                1     3   
3     1,90E+11   São Bernardo do Campo     19    M                1     3   
4     1,90E+11    São Joaquim da Barra     37    M                2     1   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO  ...  NOTA_MT  \
0                 1                1       1             0  ...    600.2   
1                 1                1       1             0  ...      NaN   
2                 1                2       3             0  ...    731.8   
3                 1                2       2             0  ...    393.2   
4                 1                1       1             0  ...      NaN   

  TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0         0                1.0  160.0  200.0  180.0  200.0  200.0   
1         1                NaN    NaN    NaN    NaN    NaN    NaN   
2         0                1.0  160.0  120.0  200.0  200.0  200.0   
3         1                1.0   80.0  160.0  120.0  100.0  100.0   
4         1                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0         B  
1           NaN         A  
2         880.0         A  
3         560.0         B  
4           NaN         B  

[5 rows x 29 columns]

##### Renomeando Registros (Linhas)

In [ ]:
dados1['RACA'] = dados1['RACA'].replace({0: 'nao_declarado', 1:'branca', 2:'preta', 3:'parda', 4:'amarela', 5:'indigina'})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO  ...  NOTA_MT  \
0                 1                1       1             0  ...    600.2   
1                 1                1       1             0  ...      NaN   
2                 1                2       3             0  ...    731.8   
3                 1                2       2             0  ...    393.2   
4                 1                1       1             0  ...      NaN   

  TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0         0                1.0  160.0  200.0  180.0  200.0  200.0   
1         1                NaN    NaN    NaN    NaN    NaN    NaN   
2         0                1.0  160.0  120.0  200.0  200.0  200.0   
3         1                1.0   80.0  160.0  120.0  100.0  100.0   
4         1                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0         B  
1           NaN         A  
2         880.0         A  
3         560.0         B  
4           NaN         B  

[5 rows x 29 columns]

In [ ]:
dados1['TP_LINGUA'] = dados1['TP_LINGUA'].replace({0:'Ingles', 1:'Espanhol'})

In [ ]:
dados1['ESCOLA'] = dados1['ESCOLA'].replace({1:'nao_respondeu', 2:'publica', 3:'privada', 4:'exterior'})

In [ ]:
dados1['INTERNET'] = dados1['INTERNET'].replace({'A':'Nao', 'B':'Sim'})

In [ ]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  nao_respondeu             0  ...   
1                 1                1  nao_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        publica             0  ...   
4                 1                1  nao_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Ingles                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Ingles                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Nao  
2         880.0       Nao  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

## Análise Exploratória e Transformação dos dados


##### Análise das idades

In [ ]:
dados1['IDADE'].value_counts()

17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
86         1
4          1
3          1
91         1
2          1
Name: IDADE, Length: 80, dtype: int64

In [ ]:
dados1['IDADE'].value_counts().sort_index()

2     1
3     1
4     1
5     2
6     2
     ..
80    2
81    3
82    2
86    1
91    1
Name: IDADE, Length: 80, dtype: int64

##### Alguns valores estranhos e curiosos: idades menores que 12 anos e maiores de 80 anos.

In [ ]:
menores_12 = dados1.query('IDADE <= 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [ ]:
# Inscrições com idade abaixo de 12 anos serão excluídos devido a alta probabilidade de erro.
menores_12

São Paulo                 5
Araçatuba                 2
Tambaú                    1
Marília                   1
Assis                     1
Auriflama                 1
Vargem Grande Paulista    1
Angatuba                  1
Santana de Parnaíba       1
Queiroz                   1
Andradina                 1
Sertãozinho               1
Embu das Artes            1
São Bernardo do Campo     1
Bálsamo                   1
Ribeirão Preto            1
Campinas                  1
Name: NO_MUNICIPIO_RESIDENCIA, dtype: int64

In [ ]:
dados_maiores_11 = dados1.loc[dados1.IDADE > 11]

In [ ]:
dados_maiores_11.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  nao_respondeu             0  ...   
1                 1                1  nao_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        publica             0  ...   
4                 1                1  nao_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Ingles                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Ingles                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Nao  
2         880.0       Nao  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

In [ ]:
dados_maiores_11['IDADE'].value_counts().sort_index()

12        6
13       29
14      355
15    11583
16    59316
      ...  
80        2
81        3
82        2
86        1
91        1
Name: IDADE, Length: 73, dtype: int64

#### Treineiros

In [ ]:
dados_maiores_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

1    93988
Name: IN_TREINEIRO, dtype: int64

In [ ]:
treineiros = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 1]

In [ ]:
treineiros.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
20      1,90E+11                  Santos     15    F                1   parda   
21      1,90E+11                Panorama     16    F                1   parda   
106     1,90E+11               São Paulo     16    F                1  branca   
109     1,90E+11        Águas de Lindóia     16    M                1  branca   
159     1,90E+11        Águas de Lindóia     15    M                1  branca   

     TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
20                  1                3  nao_respondeu             1  ...   
21                  1                3  nao_respondeu             1  ...   
106                 1                3  nao_respondeu             1  ...   
109                 1                3  nao_respondeu             1  ...   
159                 1                3  nao_respondeu             1  ...   

     NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
20     573.2    Ingles                1.0  140.0  120.0  120.0  100.0   80.0   
21       NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
106    532.6    Ingles                1.0  120.0  120.0  120.0  100.0   80.0   
109      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
159      NaN    Ingles                NaN    NaN    NaN    NaN    NaN    NaN   

     NOTA_REDACAO  INTERNET  
20          560.0       Sim  
21            NaN       Nao  
106         540.0       Sim  
109           NaN       Sim  
159           NaN       Sim  

[5 rows x 29 columns]

In [ ]:
# Salvando dados treineiros
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding= 'iso-8859-1', index = False)

In [ ]:
# Separação entre os treineiros e os vestibulandos
vestibulandos = dados_maiores_11.loc[dados_maiores_11.IN_TREINEIRO == 0]

In [ ]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  nao_respondeu             0  ...   
1                 1                1  nao_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        publica             0  ...   
4                 1                1  nao_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Ingles                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Ingles                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Nao  
2         880.0       Nao  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

In [ ]:
vestibulandos.shape

(719768, 29)

#### Análise das presenças

In [ ]:
vestibulandos['TP_PRESENCA_CN'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_CN, dtype: int64

In [ ]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_CH, dtype: int64

In [ ]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

1    533142
0    185969
2       657
Name: TP_PRESENCA_LC, dtype: int64

In [ ]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

1    500805
0    218693
2       270
Name: TP_PRESENCA_MT, dtype: int64

### RESUMO DA PRESENÇA NAS PROVAS:

* Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

* Ciências Humanas: 533142 presentes, 185969 faltaram, 657 eliminados.


* Linguagens e Códigos: 533142 presentes, 185969 faltaram, 657 eliminados.

* Matemática: 500805 presentes, 185969 faltaram, 657 eliminados.

In [ ]:
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: TP_STATUS_REDACAO, dtype: int64

### Valores Ausentes

In [ ]:
vestibulandos['IDADE'].isnull().sum()

0

In [ ]:
provas = ['NOTA_REDACAO', 'NOTA_CN', 'NOTA_MT', 'NOTA_LC', 'NOTA_CH']

In [ ]:
# Registros NAN nas notas
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [ ]:
presenca_total = ['TP_PRESENCA_CN', 'TP_PRESENCA_CH', 'TP_PRESENCA_MT', 'TP_PRESENCA_LC']

In [ ]:
# Criando uma coluna que soma as presenças nas provas
vestibulandos['presenca_total'] = vestibulandos[presenca_total].sum(axis=1)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [ ]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  nao_respondeu             0  ...   
1                 1                1  nao_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        publica             0  ...   
4                 1                1  nao_respondeu             0  ...   

   TP_LINGUA TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0     Ingles               1.0  160.0  200.0  180.0  200.0  200.0   
1   Espanhol               NaN    NaN    NaN    NaN    NaN    NaN   
2     Ingles               1.0  160.0  120.0  200.0  200.0  200.0   
3   Espanhol               1.0   80.0  160.0  120.0  100.0  100.0   
4   Espanhol               NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  presenca_total  
0         940.0       Sim               4  
1           NaN       Nao               0  
2         880.0       Nao               4  
3         560.0       Sim               4  
4           NaN       Sim               0  

[5 rows x 30 columns]

##### (variable) vestibulandos_classificados: vestibulandos

In [ ]:
vestibulandos_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
vestibulandos_classificados = vestibulandos_classificados.loc[vestibulandos.TP_PRESENCA_MT == 1]

In [ ]:
vestibulandos_classificados['TP_PRESENCA_MT'].value_counts()

1    499940
Name: TP_PRESENCA_MT, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_CH'].value_counts()

1    499940
Name: TP_PRESENCA_CH, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_CN'].value_counts()

1    499940
Name: TP_PRESENCA_CN, dtype: int64

In [ ]:
vestibulandos_classificados['TP_PRESENCA_LC'].value_counts()

1    499940
Name: TP_PRESENCA_LC, dtype: int64

In [ ]:
vestibulandos_classificados[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

##### Notas zeros na Redação

In [ ]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

0.0    9181
Name: NOTA_REDACAO, dtype: int64

In [ ]:
vestibulandos_classificados.query('COMP1 == 0')['COMP1'].value_counts()

0.0    9188
Name: COMP1, dtype: int64

In [ ]:
vestibulandos_classificados.query('COMP2 == 0')['COMP2'].value_counts()

0.0    9181
Name: COMP2, dtype: int64

In [ ]:
vestibulandos_classificados.query('COMP3 == 0')['COMP3'].value_counts()

0.0    9197
Name: COMP3, dtype: int64

In [ ]:
vestibulandos_classificados.query('COMP4 == 0')['COMP4'].value_counts()

0.0    9194
Name: COMP4, dtype: int64

In [ ]:
vestibulandos_classificados.query('COMP5 == 0')['COMP5'].value_counts()

0.0    73330
Name: COMP5, dtype: int64

## NOTAS ZEROS EM REDAÇÃO:

* COMPONENTE 1(final): 9181 notas zeros devido ao não cumprimento das regras ou tirou zero em todos os cincos componentes.
* COMPONENTE 2(Demonstrar domínio da modalidade escrita formal da Língua Portuguesa): 9188 notas zeros somente neste item.
* COMPONENTE 3(Selecionar, relacionar, organizar e interpretar informações, fatos, opiniões e argumentos em defesa de um ponto de vista): 9197 notas zeros somente neste item.
* COMPONENTE 4(Demonstrar conhecimento dos mecanismos linguísticos necessários para a construção da argumentação): 9194 notas zeros somente neste item.
* COMPONENTE 5(Elaborar proposta de intervenção para o problema abordado, respeitando os direitos humanos): 73330 notas zeros somente neste item.

In [ ]:
vestibulandos_classificados.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

0.0    343
Name: NOTA_CH, dtype: int64

In [ ]:
vestibulandos_classificados.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

0.0    36
Name: NOTA_CN, dtype: int64

In [ ]:
vestibulandos_classificados.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

0.0    135
Name: NOTA_LC, dtype: int64

In [ ]:
vestibulandos_classificados.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

0.0    56
Name: NOTA_MT, dtype: int64

In [ ]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

0.0    9181
Name: NOTA_REDACAO, dtype: int64

#### NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

* Matemática = 56
* Ciências da Natureza = 36
* Linguagens e Códigos = 135
* Ciências Humanas = 343
* Redação = 9181

In [ ]:
vestibulandos_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
5     1,90E+11              Embu-Guaçu     17    F                1   parda   
8     1,90E+11               Guarulhos     23    F                1  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  nao_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        publica             0  ...   
5                 1                2        publica             0  ...   
8                 1                1  nao_respondeu             0  ...   

   TP_LINGUA TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0     Ingles               1.0  160.0  200.0  180.0  200.0  200.0   
2     Ingles               1.0  160.0  120.0  200.0  200.0  200.0   
3   Espanhol               1.0   80.0  160.0  120.0  100.0  100.0   
5     Ingles               1.0  120.0  120.0  120.0  120.0   80.0   
8     Ingles               1.0  160.0  200.0  180.0  180.0  160.0   

   NOTA_REDACAO  INTERNET  presenca_total  
0         940.0       Sim               4  
2         880.0       Nao               4  
3         560.0       Sim               4  
5         560.0       Sim               4  
8         880.0       Sim               4  

[5 rows x 30 columns]

In [ ]:
vestibulandos_classificados = vestibulandos_classificados.drop(columns=['IN_TREINEIRO', 'presenca_total'],)

In [ ]:
vestibulandos_classificados.shape

(499940, 28)

### Salvando (Exportando) o DataFrame Tratado

In [ ]:
vestibulandos_classificados.to_csv('enem_2019_tratado.csv', encoding = 'iso-8859-1', index= False)